pkg/server/server.go

```diff
@@ -124,12 +125,18 @@ func New(conf *config.Configurations) (*BCDBHTTPServer, error) {
 
 // Start starts the server
 func (s *BCDBHTTPServer) Start() error {
-	if blockHeight, err := s.db.LedgerHeight(); err != nil {
-		return err
-	} else if blockHeight == 0 {
-		return errors.New("ledger height == 0, bootstrap failed")
-	} else {
-		s.logger.Infof("Server starting at ledger height [%d]", blockHeight)
+	for {
+		if blockHeight, err := s.db.LedgerHeight(); err != nil {
+			return err
+		} else if blockHeight == 0 {
+			// If the server is bootstrapping from the peers, it might still be trying to get the genesis block; We
+			// wait for that to happen before we start serving requests.
+			s.logger.Infof("Ledger height =0, server boostraping from peers; waiting for height >0; going to sleep for 1s...")
+			time.Sleep(time.Second)
```

Comment:
Weak guarantee. We need to let the bootstrapping process know whether the peer is being added to an existing network such that it can retry for several times before failing. 

Please create a issue to handle this cleanly. For now, I am merging this PR to fix the bug.

Reply:
The block replicator is trying and retrying forever in a different go-routine. This main thread is just holding off on starting to serve client requests until we have at least the genesis block. It is not clean... that is true.

Reply:
https://github.com/hyperledger-labs/orion-server/issues/397